README.md
# Website_TuHoang


TCPDF/CHANGELOG.TXT

6.4.2 (2021-04-02)
	- Fix PHP 8.1 type error with TCPDF_STATIC::pregSplit on preg_split
	- Fix a PHP array offset error
	- Fixed phpdoc blocks
	- Drop a PHP 4 polyfill and add a .gitattributes file

6.4.1 (2021-03-27)
	- Update tcpdf version (no code changes)

6.4.0 (2021-03-27)
	- allow styles on <HR> tags
	- check if file exists before calling unlink
	- Fix image file type for urls with query params
	- Fix SVGPath should accept 1.19.30 (equiv 1.19,.30) compacted values list
	- Fix Second parameter of TCPDF::cell() must be a number
	- PHP 8.0 function signature fixes
	- Fix vulnerability to roman numeral bombs
	- Optimized a regular expression
	- Cache file get contents calls
	- Remove mb_internal encoding handling

6.3.5 (2020-02-14)
	- Fixed curly braces in pdf417
	- Fixed a syntax error issue when accessing an index of a casted variable

6.3.4 (2020-02-12)
	- Check if imagekeys exist
	- Unlink only images in cache

6.3.3 (2020-02-12)
	- Fixed PHP 7.4 - cannot use array offset on integers
	- Fixed PDF/A-3B validation issue caused by missing pdfaSchema:property.
	- Removed backup changelog files from repo
	- Prevents the deletion of non-existent files in /tmp
	- Prevent crash in case of no list access in cache path
	- Check existence of file before delete it
	- Fixed erase users pictures
	- Fixed problem with $imagekeys undefined or unlinked
	- Fix SVGPath elliptical arc with rx/ry=0 + z should return to initial point
	- Fixed PHP 7.4 errors
	- handle integers for pages
	- Fixed background image doesn't work in RTL
	- Fixed PDF/A validity
	- Fixed datamatrix.php for PHP 7.4
	- Fixed deprecated PHP features

6.3.2 (2019-09-20)
	- Update ICC profile

6.3.1 (2019-09-20)
	- Fix reported version
	- Fix Undefined property: GLPIPDF::$imagekeys

6.3.0 (2019-09-19)
	- fix SpotColor handling in HTML
	- Add an additional empty test to prevent error in PHP 7.2
	- Fix the documentation how to calculate the cell height
	- Drop duplicated array indices
	- Fix TCPDF_STATIC::fileGetContents()
	- Introduce other version of pdfA (2 and 3)
	- Add UF and AFRelationship missing
	- Fix performance issue of cloned instances
	- Change glob to readdir which performs better
	- URI in PDF can result in E_NOTICE
	- Fix a warning for PHP 7.4
	- Fixed gradient offsets for percentage-based stops.
	- Fixed file_get_contents return value should also be checked for a non-empty string
	- Fix Array and string offset access syntax with curly braces is deprecated
	- Fix PHP Warning: chr() expects parameter 1 to be int
	- Add a VERSION file

6.2.26 (2018-10-16)
	- Update sRGB.icc with the one from the Debian package icc-profiles-free
	- Fix unsupported operand types error when codepoints arrays are merged

6.2.25 (2018-09-23)
	- Fix support for image URLs.

6.2.24
	- Support remote urls when checking if file exists.

6.2.23 (2018-09-22)
	- Simplify file_exists function.

6.2.22 (2018-09-14)
	- Fixes on `include/tcpdf_images.php`, `include/tcpdf_static.php` and `tcpdf.php` about file handling

6.2.21 (2018-09-14)
	- _no code changes_

6.2.20 (2018-09-14)
	- Fix for security vulnerability: Using the phar:// wrapper it was possible to trigger the unserialization of user provided data.

6.2.19 (2018-09-14)
	- Merge various fixes for PHP 7.3 compatibility and security.

6.2.13 (2016-06-10)
	- IMPORTANT: A new version of this library is under development at https://github.com/tecnickcom/tc-lib-pdf and as a consequence this version will not receive any additional development or support. This version should be considered obsolete, new projects should use the new version as soon it will become stable.

6.2.12 (2015-09-12)
	- fix composer package name to tecnickcom/tcpdf

6.2.11 (2015-08-02)
	- Bug #1070 "PNG regression in 6.2.9 (they appear as their alpha channel)" was fixed.
	- Bug #1069 "Encoded SRC URLs in <img> tags don't work anymore" was fixed.

6.2.10 (2015-07-28)
	- Minor mod to PNG parsing.
	- Make dependency on mcrypt optional.

6.2.8 (2015-04-29)
	- Removed unwanted file.

6.2.7 (2015-04-28)
	- Merged PR 17: Avoid warning when iterating a non-array variable.
	- Merged PR 16: Improve MuliCell param definition.
	- Improved column check (PR 15).
	- Merged PR 11: Use stream_is_local instead of limit to file://.
	- Merged PR 10: ImageMagick link on README.txt.

6.2.6 (2015-01-28)
	- Bug #1008 "UTC offset sing breaks PDF/A-1b compliance" was fixed.

6.2.5 (2015-01-24)
	- Bug #1019 "$this in static context" was fixed.
	- Bug #1015 "Infinite loop in getIndirectObject method of parser" was fixed.

6.2.4 (2015-01-08)
	- fix warning related to empty K_PATH_URL.
	- fix error when a $table_colwidths key is not set.

6.2.3 (2014-12-18)
	- New comment.
	- Moved the K_PATH_IMAGES definition in tcpdf_autoconfig.

6.2.2 (2014-12-18)
	- Fixed mispelled words.
	- Fixed version number.

6.2.1 (2014-12-18)
	- The constant K_TCPDF_THROW_EXCEPTION_ERROR is now set to false in the default configuration file.
	- An issue with the _destroy() method was fixed.

6.2.0 (2014-12-10)
	- Bug #1005 "Security Report, LFI posting internal files externally abusing default parameter" was fixed.
	- Static methods serializeTCPDFtagParameters() and unserializeTCPDFtagParameters() were moved as non static to the main TCPDF class (see changes in example n. 49).
	- Deprecated methods were removed, please use the equivalents defined in other classes (i.e. TCPDF_STATIC and TCPDF_FONTS).
	- The constant K_TCPDF_CALLS_IN_HTML is now set by default to FALSE.
	- DLE, DLX and DLP page format was added.
	- Page format are now defined as a public property in TCPDF_STATIC.

6.1.1 (2014-12-09)
	- Fixed bug with the register_shutdown_function().

6.1.0 (2014-12-07)
	- The method TCPDF_STATIC::getRandomSeed() was improved.
	- The disk caching feature was removed.
	- Bug #1003 "Backslashes become duplicated in table, using WriteHTML" was fixed.
	- Bug #1002 "SVG radialGradient within non-square Rect" was fixed.

6.0.099 (2014-11-15)

TCPDF/tcpdf_autoconfig.php
<?php
//============================================================+
// File name   : tcpdf_autoconfig.php
// Version     : 1.1.1
// Begin       : 2013-05-16
// Last Update : 2014-12-18
// Authors     : Hoàng Mạnh Tú LTD - www.tecnick.com - info@tecnick.com
// License     : GNU-LGPL v3 (http://www.gnu.org/copyleft/lesser.html)
// -------------------------------------------------------------------
// Copyright (C) 2011-2014 Hoàng Mạnh Tú LTD
//
// This file is part of TCPDF software library.
//
// TCPDF is free software: you can redistribute it and/or modify it
// under the terms of the GNU Lesser General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// TCPDF is distributed in the hope that it will be useful, but
// WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
// See the GNU Lesser General Public License for more details.
//
// You should have received a copy of the License
// along with TCPDF. If not, see
// <http://www.tecnick.com/pagefiles/tcpdf/LICENSE.TXT>.
//
// See LICENSE.TXT file for more information.
// -------------------------------------------------------------------
//
// Description : Try to automatically configure some TCPDF
//               constants if not defined.
//
//============================================================+

/**
 * @file
 * Try to automatically configure some TCPDF constants if not defined.
 * @package com.tecnick.tcpdf
 * @version 1.1.1
 */

// DOCUMENT_ROOT fix for IIS Webserver
if ((!isset($_SERVER['DOCUMENT_ROOT'])) OR (empty($_SERVER['DOCUMENT_ROOT']))) {
	if(isset($_SERVER['SCRIPT_FILENAME'])) {
		$_SERVER['DOCUMENT_ROOT'] = str_replace( '\\', '/', substr($_SERVER['SCRIPT_FILENAME'], 0, 0-strlen($_SERVER['PHP_SELF'])));
	} elseif(isset($_SERVER['PATH_TRANSLATED'])) {
		$_SERVER['DOCUMENT_ROOT'] = str_replace( '\\', '/', substr(str_replace('\\\\', '\\', $_SERVER['PATH_TRANSLATED']), 0, 0-strlen($_SERVER['PHP_SELF'])));
	} else {
		// define here your DOCUMENT_ROOT path if the previous fails (e.g. '/var/www')
		$_SERVER['DOCUMENT_ROOT'] = '/';
	}
}
$_SERVER['DOCUMENT_ROOT'] = str_replace('//', '/', $_SERVER['DOCUMENT_ROOT']);
if (substr($_SERVER['DOCUMENT_ROOT'], -1) != '/') {
	$_SERVER['DOCUMENT_ROOT'] .= '/';
}

// Load main configuration file only if the K_TCPDF_EXTERNAL_CONFIG constant is set to false.
if (!defined('K_TCPDF_EXTERNAL_CONFIG') OR !K_TCPDF_EXTERNAL_CONFIG) {
	// define a list of default config files in order of priority
	$tcpdf_config_files = array(dirname(__FILE__).'/config/tcpdf_config.php', '/etc/php-tcpdf/tcpdf_config.php', '/etc/tcpdf/tcpdf_config.php', '/etc/tcpdf_config.php');
	foreach ($tcpdf_config_files as $tcpdf_config) {
		if (@file_exists($tcpdf_config) AND is_readable($tcpdf_config)) {
			require_once($tcpdf_config);
			break;
		}
	}
}

if (!defined('K_PATH_MAIN')) {
	define ('K_PATH_MAIN', dirname(__FILE__).'/');
}

if (!defined('K_PATH_FONTS')) {
	define ('K_PATH_FONTS', K_PATH_MAIN.'fonts/');
}

if (!defined('K_PATH_URL')) {
	$k_path_url = K_PATH_MAIN; // default value for console mode
	if (isset($_SERVER['HTTP_HOST']) AND (!empty($_SERVER['HTTP_HOST']))) {
		if(isset($_SERVER['HTTPS']) AND (!empty($_SERVER['HTTPS'])) AND (strtolower($_SERVER['HTTPS']) != 'off')) {
			$k_path_url = 'https://';
		} else {
			$k_path_url = 'http://';
		}
		$k_path_url .= $_SERVER['HTTP_HOST'];
		$k_path_url .= str_replace( '\\', '/', substr(K_PATH_MAIN, (strlen($_SERVER['DOCUMENT_ROOT']) - 1)));
	}
	define ('K_PATH_URL', $k_path_url);
}

if (!defined('K_PATH_IMAGES')) {
	$tcpdf_images_dirs = array(K_PATH_MAIN.'examples/images/', K_PATH_MAIN.'images/', '/usr/share/doc/php-tcpdf/examples/images/', '/usr/share/doc/tcpdf/examples/images/', '/usr/share/doc/php/tcpdf/examples/images/', '/var/www/tcpdf/images/', '/var/www/html/tcpdf/images/', '/usr/local/apache2/htdocs/tcpdf/images/', K_PATH_MAIN);
	foreach ($tcpdf_images_dirs as $tcpdf_images_path) {
		if (@file_exists($tcpdf_images_path)) {
			define ('K_PATH_IMAGES', $tcpdf_images_path);
			break;
		}
	}
}

if (!defined('PDF_HEADER_LOGO')) {
	$tcpdf_header_logo = '';
	if (@file_exists(K_PATH_IMAGES.'tcpdf_logo.jpg')) {
		$tcpdf_header_logo = 'tcpdf_logo.jpg';
	}
	define ('PDF_HEADER_LOGO', $tcpdf_header_logo);
}

if (!defined('PDF_HEADER_LOGO_WIDTH')) {
	if (!empty($tcpdf_header_logo)) {
		define ('PDF_HEADER_LOGO_WIDTH', 30);
	} else {
		define ('PDF_HEADER_LOGO_WIDTH', 0);
	}
}

if (!defined('K_PATH_CACHE')) {
	$K_PATH_CACHE = ini_get('upload_tmp_dir') ? ini_get('upload_tmp_dir') : sys_get_temp_dir();
	if (substr($K_PATH_CACHE, -1) != '/') {
		$K_PATH_CACHE .= '/';
	}
	define ('K_PATH_CACHE', $K_PATH_CACHE);
}

if (!defined('K_BLANK_IMAGE')) {
	define ('K_BLANK_IMAGE', '_blank.png');
}

if (!defined('PDF_PAGE_FORMAT')) {
	define ('PDF_PAGE_FORMAT', 'A4');
}

if (!defined('PDF_PAGE_ORIENTATION')) {
	define ('PDF_PAGE_ORIENTATION', 'P');
}

if (!defined('PDF_CREATOR')) {
	define ('PDF_CREATOR', 'TCPDF');
}

if (!defined('PDF_AUTHOR')) {
	define ('PDF_AUTHOR', 'TCPDF');
}

if (!defined('PDF_HEADER_TITLE')) {
	define ('PDF_HEADER_TITLE', 'Showroom Tú Hoàng');
}

if (!defined('PDF_HEADER_STRING')) {
	define ('PDF_HEADER_STRING', "by Hoàng Mạnh Tú\nwww.localhost/Website_Tu_Hoang/index.php");
}

if (!defined('PDF_UNIT')) {
	define ('PDF_UNIT', 'mm');
}

if (!defined('PDF_MARGIN_HEADER')) {
	define ('PDF_MARGIN_HEADER', 5);
}

if (!defined('PDF_MARGIN_FOOTER')) {
	define ('PDF_MARGIN_FOOTER', 10);
}

if (!defined('PDF_MARGIN_TOP')) {
	define ('PDF_MARGIN_TOP', 27);
}

if (!defined('PDF_MARGIN_BOTTOM')) {
	define ('PDF_MARGIN_BOTTOM', 25);
}

if (!defined('PDF_MARGIN_LEFT')) {
	define ('PDF_MARGIN_LEFT', 15);
}

if (!defined('PDF_MARGIN_RIGHT')) {
	define ('PDF_MARGIN_RIGHT', 15);
}

if (!defined('PDF_FONT_NAME_MAIN')) {
	define ('PDF_FONT_NAME_MAIN', 'helvetica');
}

if (!defined('PDF_FONT_SIZE_MAIN')) {
	define ('PDF_FONT_SIZE_MAIN', 10);
}

if (!defined('PDF_FONT_NAME_DATA')) {
	define ('PDF_FONT_NAME_DATA', 'helvetica');
}

if (!defined('PDF_FONT_SIZE_DATA')) {
	define ('PDF_FONT_SIZE_DATA', 8);
}

if (!defined('PDF_FONT_MONOSPACED')) {
	define ('PDF_FONT_MONOSPACED', 'courier');
}

if (!defined('PDF_IMAGE_SCALE_RATIO')) {
	define ('PDF_IMAGE_SCALE_RATIO', 1.25);
}

if (!defined('HEAD_MAGNIFICATION')) {
	define('HEAD_MAGNIFICATION', 1.1);
}

if (!defined('K_CELL_HEIGHT_RATIO')) {
	define('K_CELL_HEIGHT_RATIO', 1.25);
}

if (!defined('K_TITLE_MAGNIFICATION')) {
	define('K_TITLE_MAGNIFICATION', 1.3);
}

if (!defined('K_SMALL_RATIO')) {
	define('K_SMALL_RATIO', 2/3);
}

if (!defined('K_THAI_TOPCHARS')) {
	define('K_THAI_TOPCHARS', true);
}

if (!defined('K_TCPDF_CALLS_IN_HTML')) {
	define('K_TCPDF_CALLS_IN_HTML', false);
}

if (!defined('K_TCPDF_THROW_EXCEPTION_ERROR')) {
	define('K_TCPDF_THROW_EXCEPTION_ERROR', false);
}

if (!defined('K_TIMEZONE')) {
	define('K_TIMEZONE', @date_default_timezone_get());
}

//============================================================+
// END OF FILE
//============================================================+
